scripts/forecast_csv.py

```diff
@@ -0,0 +1,110 @@
+import os
+import pandas as pd
+from datetime import datetime, timedelta
+from quartz_solar_forecast.forecast import run_forecast
+from quartz_solar_forecast.pydantic_models import PVSite
+import unittest
+from unittest.mock import patch
+from datetime import datetime
+
+def generate_forecast(init_time_freq, start_datetime, end_datetime, site_name, latitude, longitude, capacity_kwp):
+    """
+    Generates forecasts at specified intervals and saves them into a CSV file.
+
+    Args:
+        init_time_freq (int): The frequency in hours at which the forecasts are generated.
+        start_datetime (str): The starting date and time for generating forecasts.
+        end_datetime (str): The ending date and time for generating forecasts.
+        site_name (str): The name of the site for which the forecasts are generated.
+        latitude (float): The latitude of the PV site.
+        longitude (float): The longitude of the PV site.
+        capacity_kwp (float): The capacity of the PV site in kilowatts peak (kWp).
+    """
+    start = datetime.strptime(start_datetime, "%Y-%m-%d %H:%M:%S")
+    start_date = start.date()
+    end = datetime.strptime(end_datetime, "%Y-%m-%d %H:%M:%S")
+    end_date = end.date()
+    all_forecasts = pd.DataFrame()
+    site = PVSite(latitude=latitude, longitude=longitude, capacity_kwp=capacity_kwp)
+
+    init_time = start
+    while init_time <= end:
+        print(f"Running forecast for initialization time: {init_time}")
+        predictions_df = run_forecast(site=site, ts=init_time.strftime("%Y-%m-%d %H:%M:%S"))
+        predictions_df['forecast_init'] = init_time
+        all_forecasts = pd.concat([all_forecasts, predictions_df])
+        init_time += timedelta(hours=init_time_freq)
+    
+    output_dir = os.path.join(os.getcwd(), 'csv_forecasts')
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)  
+    output_file_name = f"forecast_{site_name}_{start_date}_{end_date}.csv"
+    output_file_path = os.path.join(output_dir, output_file_name)
+
+    all_forecasts.to_csv(output_file_path, index=False)
```

Comment:
You are loosing the index (which is the time in which the forecast is made) when saving. I would suggest reseting the index before you save so that you don't loose this information.